### PR TITLE
bugfix(ga/ga360): nonInteraction can be 0 for enhancedEcoomerce event…

### PR DESCRIPTION
**Pull request by shrouti1507**
nonInteraction can be 0 for enhancedEcoomerce events, if resides as message.properties.nonInteraction.

## Description of the change

> Description here

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

**Review by utsabc — APPROVED** [review]

**Comment by utsabc** [comment]
@ItsSudip can you review also

File: v0/destinations/ga/transform.js [hunk 1]
```diff
@@ -1017,6 +1021,8 @@ function processSingleMessage(message, destination) {
         .build();
   }
 
+  console.log(JSON.stringify(customParams));
```
Comment (ItsSudip) [inline v0/destinations/ga/transform.js:1024]: Why there is a `console.log`?

File: v0/destinations/ga/transform.js [hunk 2]
```diff
@@ -929,7 +929,11 @@ function processSingleMessage(message, destination) {
           ? nameToEventMap[eventName].category
           : ConfigCategory.NON_ECOM;
         category.hitType = "event";
-        customParams.ni = 1;
+        customParams.ni =
+          message.properties !== undefined &&
+          message.properties.nonInteraction !== undefined
+            ? message.properties.nonInteraction
+            : 1;
```
Comment (ItsSudip) [inline v0/destinations/ga/transform.js:936]: We can use conditional chaining here.

File: v0/destinations/ga/transform.js (hunk 1 above)
Comment (shrouti1507) [reply v0/destinations/ga/transform.js]: removed

File: v0/destinations/ga/transform.js (hunk 2 above)
Comment (shrouti1507) [reply v0/destinations/ga/transform.js]: conditional chaining is generating 1 for nonInteraction set as 0. Removed the first condition and used conditional chaining for that

**Review by utsabc — APPROVED** [review]

**Review by ItsSudip — APPROVED** [review]

**Pull request merged**